### PR TITLE
feat: call vault server to get doc meta

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

**Review by rolsonquadras — APPROVED** [review]

File: pkg/restapi/comparator/operation/compare.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+				respondErrorf(w, http.StatusInternalServerError, "failed to get doc meta: %s", err.Error())
+
+				return
+			}
+
+			logger.Infof(docMeta.URI)
+
+			// TODO where to put docMeta URI in doc query struct for doc1
+
+			// TODO how to get query id for doc2 from orgCompQueryZCAP
```
Comment (llorllale) [inline pkg/restapi/comparator/operation/compare.go:35]: Will be the invocation target on the zcap.

File: pkg/restapi/comparator/operation/compare.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+			docMeta, err := o.vaultClient.GetDocMetaData(*q.DocID, *q.VaultID)
+			if err != nil {
+				respondErrorf(w, http.StatusInternalServerError, "failed to get doc meta: %s", err.Error())
+
+				return
+			}
+
+			logger.Infof(docMeta.URI)
+
+			// TODO where to put docMeta URI in doc query struct for doc1
```
Comment (llorllale) [inline pkg/restapi/comparator/operation/compare.go:33]: Do you mean [this](https://github.com/trustbloc/edge-service/blob/8443964417f1e806286514030bb40edf28456e2a/pkg/restapi/csh/operation/openapi/doc_query.go#L22-L39)?

File: pkg/restapi/comparator/operation/openapi/authorization.go [hunk 3]
```diff
@@ -0,0 +1,276 @@
+// Code generated by go-swagger; DO NOT EDIT.
```
Comment (llorllale) [inline pkg/restapi/comparator/operation/openapi/authorization.go:1]: Looks like these models were generated from the wrong spec - should be from [this one](https://github.com/trustbloc/edge-service/blob/0539d17f92353cedb99687e5ad34369a6a4ff90f/cmd/comparator-rest/docs/openapi.yaml).

File: pkg/restapi/comparator/operation/compare.go (hunk 2 above)
Comment (fqutishat) [reply pkg/restapi/comparator/operation/compare.go:33]: Yes

File: pkg/restapi/comparator/operation/openapi/authorization.go (hunk 3 above)
Comment (fqutishat) [reply pkg/restapi/comparator/operation/openapi/authorization.go:1]: done

**Review by soluchok — APPROVED** [review]

File: cmd/comparator-rest/docs/openapi.yaml [hunk 4]
```diff
@@ -69,43 +69,20 @@ paths:
...
-                    }
-                  },
-                  {
-                    "type": "auth",
-                    "authToken": "123456789abcdefghi"
-                  }
-                ]
-              }
-            }
+              $ref: "#/components/schemas/ComparisonRequest"
```
Comment (llorllale) [inline cmd/comparator-rest/docs/openapi.yaml:78]: There doesn't seem to be any practical need for this refactor.

File: cmd/comparator-rest/docs/openapi.yaml [hunk 5]
```diff
@@ -69,43 +69,20 @@ paths:
...
       tags:
         - required
+      description: Evaluates an operator with its inputs and returns the result.
       requestBody:
         required: true
         content:
           application/json:
             schema:
-              $ref: "#/components/schemas/Comparison"
-            example: {
```
Comment (llorllale) [inline cmd/comparator-rest/docs/openapi.yaml:78]: Please don't delete the examples - they're there to illustrate usage of the API.

File: cmd/comparator-rest/docs/openapi.yaml [hunk 6]
```diff
@@ -263,9 +246,12 @@ components:
         - $ref: "#/components/schemas/Query"
         - type: object
           required:
+            - vaultID
             - docID
             - authTokens
           properties:
+            vaultID:
```
Comment (llorllale) [inline cmd/comparator-rest/docs/openapi.yaml:253]: I agree with this addition.

File: cmd/comparator-rest/docs/openapi.yaml [hunk 7]
```diff
@@ -218,11 +195,17 @@ components:
...
               type: integer
               description: Duration (in seconds) for which this authorization will remain valid.
-    Comparison:
+    ComparisonRequest:
       type: object
       properties:
         op:
           $ref: "#/components/schemas/Operator"
+    Comparison:
+      description: TODO - "comparison" does not sound apt as a name
```
Comment (llorllale) [inline cmd/comparator-rest/docs/openapi.yaml:204]: This description needs to be fixed: `Result of the comparison.`

File: cmd/comparator-rest/docs/openapi.yaml (hunk 4 above)
Comment (fqutishat) [reply cmd/comparator-rest/docs/openapi.yaml]: done

File: cmd/comparator-rest/docs/openapi.yaml (hunk 5 above)
Comment (fqutishat) [reply cmd/comparator-rest/docs/openapi.yaml]: done

File: cmd/comparator-rest/docs/openapi.yaml (hunk 7 above)
Comment (fqutishat) [reply cmd/comparator-rest/docs/openapi.yaml]: done

**Pull request merged**